src/pages/ListSwaps/SwapRow.tsx

```diff
@@ -40,208 +29,52 @@ function AssetCell({ asset }: AssetCellProps) {
   }
 }
 
-enum DialogState {
-  Closed,
-  CommunicationDialogOpen,
-  LedgerParamsDialogOpen,
-  LedgerDialogOpen
-}
-
-interface LedgerActionSpec {
-  key: string;
-  label: string;
-}
-
-function actionQueryParams(swap: Swap, actionName: string): LedgerActionSpec[] {
-  const ledger = actionLedger(swap, actionName);
-
-  if (!ledger) {
-    return [];
+const useStyles = makeStyles(() => ({
+  tableRow: {
+    cursor: "pointer"
   }
+}));
 
-  if (
-    ledger.name === "bitcoin" &&
-    (actionName === "redeem" || actionName === "refund")
-  ) {
-    return [
-      {
-        key: "address",
-        label: "Bitcoin address"
-      },
-      { key: "fee_per_byte", label: "Fee per byte" }
-    ];
-  } else {
-    return [];
-  }
+interface SwapRowProps extends RouteComponentProps {
+  swap: Swap;
 }
 
-interface LedgerParamsField {
-  name: string;
-  value: string;
-}
+function SwapRow({ swap, history }: SwapRowProps) {
+  const classes = useStyles();
+  const swapLink = swap.links.filter(link => link.rel[0] === "self")[0].href;
```

Comment:
```suggestion
  const swapLink = swap.links.find(link => link.rel.contains("self)).href;
```

Reply:
This is cleaner, but `find` returns the first element that matches the predicate or undefined. Do we want to handle that case or is it safe to assume that the comit node's response will always include a `self` link?

Reply:
Currently, you are `[0]` the result of `filter`, which might yield an array with 0 elements, so that behaviour should stay the same? :)
If we don't get a self-link, that is a bug in the comit node.

Reply:
I agree. The thing is the compiler complains about the result of `swap.links.find(link => link.rel.includes("self)` being possibly undefined, whereas before it didn't because it can't reason about array length.

Reply:
I think for now, we can fail, if we don't have a self link. It is not nice but the better solution (disable click handler and derive a `key` in a different, deterministic manner) is probably an overkill for now, given that it is definitely a bug in the comit node if we don't have a self link.